### PR TITLE
updated notebooks to use latest langchain version

### PR DESCRIPTION
**Pull request by josefinarcasanova**
Content changes:

- Improved general writing
- Standardized environment setup sections
- Added my name to authors, as editor
- Added more references to LC docs

Code changes:

- Updated to latest langchain version
- Changed LLMChain and SimpleSequentialChain (deprecated) to LCEL Runnable flows

### CODE REVIEW

**Review by krab7191 — CHANGES_REQUESTED** [review]
Reviewing now. Can you just format the JSON so it's easier to see diffs and maintain ?

**Review by krab7191 — COMMENTED** [review]
Let's also remove the cell output

**Review by krab7191 — COMMENTED** [review]
I'm using python 3.13, looks like this causes an issue with pandas install. Lets make a note of that. https://stackoverflow.com/a/78719227/18236359

File: content/labs/Watsonx/WatsonxAI/files/203/LangChain_Integration.ipynb [hunk 1]
```diff
@@ -1,576 +1 @@
...
...:{"pycharm":{"name":"#%% md\n"}},"source":["<a id=\"summary\"></a>\n","## Summary and next steps\n","\n"," You successfully completed this notebook!.\n"," \n"," You learned how to use chain using custom llm WastonxLLM.\n"," \n","Check out our _[Online Documentation](https://ibm.github.io/watson-machine-learning-sdk/samples.html)_ for more samples, tutorials, documentation, how-tos, and blog posts. "]},{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%% md\n"}},"source":["### Authors: \n"," - **Mateusz Szewczyk**, Software Engineer at Watson Machine Learning.\n"," - **Josefina Casanova**, Engagement Lead, Build Lab Americas. Edited for L4 watsonx course. 2024"]},{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%% md\n"}},"source":["Copyright © 2024 IBM. This notebook and its source code are released under the terms of the MIT License."]}],"metadata":{"kernelspec":{"display_name":"Python 3.11","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.11.9"}},"nbformat":4,"nbformat_minor":4}
```
[the remark this answers is not in the document]
Comment (krab7191) [reply content/labs/Watsonx/WatsonxAI/files/203/LangChain_Integration.ipynb:1]: Confirmed working locally with python3.11. See comment below. Let's just make a note in the notebook for users who want to run locally.

File: content/labs/Watsonx/WatsonxAI/files/203/LangChain_Memory.ipynb [hunk 2]
```diff
@@ -1,304 +1 @@
...
...                       Numbered list of complaints:', additional_kwargs={}, response_metadata={}), AIMessage(content=' 1. Credit report errors, 2. Identity theft, 3. Failure to correct errors in a timely manner.', additional_kwargs={}, response_metadata={}), HumanMessage(content='Does the numbered list of complaints contain a statement about identity fraud? Provide a short answer: yes or no.', additional_kwargs={}, response_metadata={}), AIMessage(content='**\\n\\nYes', additional_kwargs={}, response_metadata={})])}\n"]}],"source":["# For debugging, print the history of the conversation\n","print(local_memory)"]},{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%% md\n"}},"source":["### Authors: \n","- **Elena Lowery**, Data and AI Architect\n","- **Josefina Casanova**, Engagement Lead, Build Lab Americas. Edited for L4 watsonx course. 2024"]}],"metadata":{"kernelspec":{"display_name":"Python 3.11","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.11.9"}},"nbformat":4,"nbformat_minor":4}
```
Comment (krab7191) [inline content/labs/Watsonx/WatsonxAI/files/203/LangChain_Memory.ipynb:1]: For memory we'll need another package: `!{sys.executable} -m pip install langchain-community | tail -n 1`

Comment (krab7191) [reply content/labs/Watsonx/WatsonxAI/files/203/LangChain_Memory.ipynb:1]: Might want to look at formatting some of the output in this file too.

File: content/labs/Watsonx/WatsonxAI/files/203/LangChain_Simple_RAG.ipynb [hunk 3]
```diff
@@ -1,253 +1 @@
...
... don't try to make up an answer.\"}"]},"execution_count":7,"metadata":{},"output_type":"execute_result"}],"source":["# Test answering questions based on the provided .pdf file\n","question = \"What is Generative AI?\"\n","# question = \"What does it take to build a generative AI model?\"\n","# question = \"What are the limitations of generative AI models?\"\n","file_path = \"https://raw.githubusercontent.com/CloudPak-Outcomes/Outcomes-Projects/main/L4assets/watsonx.ai-Assets/Documents/Generative_AI_Overview.pdf\"\n","\n","answer_questions_from_doc(\n","    file_path = file_path,\n","    question = question,\n","    print_res = True\n",")"]},{"cell_type":"markdown","metadata":{"id":"7868d367-ecc9-4320-98ea-c8aa33a5ff96"},"source":["\n","### Authors:\n","- **Josefina Casanova**, Engagement Lead, Build Lab Americas. Edited for L4 watsonx course. 2024"]}],"metadata":{"kernelspec":{"display_name":"Python 3.11","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.11.9"}},"nbformat":4,"nbformat_minor":4}
```
Comment (krab7191) [inline content/labs/Watsonx/WatsonxAI/files/203/LangChain_Simple_RAG.ipynb:1]: I think we need to change to `from langchain_community.embeddings import HuggingFaceEmbeddings` from `langchain_huggingface`

**Comment by josefinarcasanova** [comment]
> Reviewing now. Can you just format the JSON so it's easier to see diffs and maintain ?

Sorry, not sure how to. Googled it quickly and didn't find a fix. Perhaps you know? I'm working the `ipynb` directly in VSCode.

File: content/labs/Watsonx/WatsonxAI/files/203/LangChain_Integration.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (josefinarcasanova) [reply content/labs/Watsonx/WatsonxAI/files/203/LangChain_Integration.ipynb:1]: Done. Added environment setup explanation for wx a non-wx and considerations section to explain 3.13 non-support.

File: content/labs/Watsonx/WatsonxAI/files/203/LangChain_Memory.ipynb (hunk 2 above)
Comment (josefinarcasanova) [reply content/labs/Watsonx/WatsonxAI/files/203/LangChain_Memory.ipynb:1]: Done, added package and clarified as with the other notebooks. 

As in output you mean clearing it for export, or that doesn't display right?

**Comment by krab7191** [comment]
> > Reviewing now. Can you just format the JSON so it's easier to see diffs and maintain ?
> 
> Sorry, not sure how to. Googled it quickly and didn't find a fix. Perhaps you know? I'm working the `ipynb` directly in VSCode.

The way I do it in Vscode is use the command palette -> `view: reopen editor with` -> `text editor` and then format the json.

**Comment by josefinarcasanova** [comment]
> > > Reviewing now. Can you just format the JSON so it's easier to see diffs and maintain ?
> > 
> > 
> > Sorry, not sure how to. Googled it quickly and didn't find a fix. Perhaps you know? I'm working the `ipynb` directly in VSCode.
> 
> The way I do it in Vscode is use the command palette -> `view: reopen editor with` -> `text editor` and then format the json.

thank you for the instructions! I have formatted the JSON as requested.

Workshop is next week, would like to solve this PR before then. Are my changes ok? Something else I need to look into?

**Review by krab7191 — APPROVED** [review]

**Pull request merged**